target/ppc/translate/fixedpoint-impl.c.inc

```diff
@@ -70,6 +70,63 @@ static bool do_ldst_X(DisasContext *ctx, arg_X *a, bool update,
     return do_ldst(ctx, a->rt, a->ra, cpu_gpr[a->rb], update, store, mop);
 }
 
+static bool do_ldst_quad(DisasContext *ctx, int ra, TCGv tgt_a, TCGv tgt_b,
+                         TCGv displ, bool store)
+{
+    TCGv ea;
+
+    gen_set_access_type(ctx, ACCESS_INT);
+    ea = tcg_temp_new();
+    do_ea_calc(ctx, ra, displ, ea);
+
+    if (tb_cflags(ctx->base.tb) & CF_PARALLEL) {
+        if (HAVE_ATOMIC128) {
+            TCGv_i32 oi = tcg_temp_new_i32();
+            tcg_gen_movi_i32(oi, make_memop_idx(MO_Q, ctx->mem_idx));
+            if (store) {
+                if (ctx->le_mode) {
```

Comment:
This if here has me a bit confused. You already check the endianness before calling do_ldst_quad and change the order of the parameters when calling it. By checking it again here, you undo what the last if did... shouldn't there be only one if in one of the functions?

You should probably only check the endianness here, since you call different functions, and flip the parameters accordingly only here, since its the only difference in the calls

Reply:
The first check is concerned with the non-atomic path of `do_ldst_quad`, to allow code sharing with the prefixed load/store (that are endian-independent wrt even/odd register). The atomic path, however, can't ignore endianness, because the helpers are different, so we need to check it again.

Reply:
I mean in the code in line 209 and 213 for example. Can't you have a single call there? (maybe this wasn't the best part to ask this...)

Reply:
Do you mean the two `return`s in `trans_LQ`/`trans_STQ`? They make the lower part of the quadword to be loaded into the odd register. This is something that changes between prefixed and non-prefixed load/store, so it must be outside of `do_ldst_quad`.